src/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -156,41 +156,124 @@ public Task<Stream> GetStreamAsync(string requestUri)
         // Unbuffered by default
         public Task<Stream> GetStreamAsync(Uri requestUri)
         {
-            return GetContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, Stream.Null,
+            return GetStreamContentAsync(requestUri, HttpCompletionOption.ResponseHeadersRead,
                 content => content.ReadAsStreamAsync());
         }
 
-        private Task<T> GetContentAsync<T>(Uri requestUri, HttpCompletionOption completionOption, T defaultValue,
-            Func<HttpContent, Task<T>> readAs)
+        private Task<string> GetStringContentAsync(Uri requestUri, HttpCompletionOption completionOption, 
+            Func<HttpContent, Task<string>> readAs)
         {
-            TaskCompletionSource<T> tcs = new TaskCompletionSource<T>();
+            var tcs = new TaskCompletionSource<string>(readAs);
 
-            GetAsync(requestUri, completionOption).ContinueWithStandard(requestTask =>
+            GetAsync(requestUri, completionOption).ContinueWithStandard(tcs, (requestTask, state) =>
             {
-                if (HandleRequestFaultsAndCancelation(requestTask, tcs))
+                var innerTcs = (TaskCompletionSource<string>)state;
+                var innerReadAs = (Func<HttpContent, Task<string>>)innerTcs.Task.AsyncState;
+                if (HandleRequestFaultsAndCancelation(requestTask, innerTcs))
                 {
                     return;
                 }
                 HttpResponseMessage response = requestTask.Result;
                 if (response.Content == null)
                 {
-                    tcs.TrySetResult(defaultValue);
+                    innerTcs.TrySetResult(string.Empty);
```

Comment:
I expect it's relatively uncommon for the content to be null?  Seems a bit unfortunate that we appear to be triplicating this whole method primarily due to needing a different default value here for each of the diferent types.  I haven't tried this, but alternative could be to just get a value based on the type being used and keeping a single implementation, e.g.
```C#
T result = 
    typeof(T) == typeof(string) ? (T)(object)string.Empty :
    typeof(T) == typeof(Stream) ? (T)(object)Stream.Null :
    typeof(T) == typeof(byte[]) ? (T)(object)Array.Empty<byte>() :
    default(T);
innerTcs.TrySetResult(result);
```
Alternatively, a static dictionary from type to default value could be maintained, prepopulated with all of these defaults.